### ISSUE TITLE
Search auto closes on Android

### ISSUE THREAD

**Issue by khawkins98** [select2/select2#2061]
Quick note: We're using a very modified version of Select2, so I don't think this should go upstream, but I'm sharing it for posterity.

Our problem was with the Android browser, it appeared to be closing our Select2 search in a way that was different from past issues. 

Symptom:
- Using latest Select2 3.4.5 (and recent versions 3.4.x)
- Browser is native Android (reports were from Android 4.x; Chrome is fine)
- User taps search field
- Select2 interface shows
- 10 or 20 ms later, the Android keyboard shows, select 2 closes
- User can type on keyboard and search works fine

A user workaround was to long tap on the search field, and Select2 works properly.

We worked around this in the code by using a timer, in the style of:

At the start of select2.js:
```
    var openTime = new Date().getTime();
```

At the end of opening: function
```
    openTime = new Date().getTime();
```

At the start of close: function
```
            // if it's been open for less than one second, don't exit
            var tempCounter = new Date().getTime();
            if (openTime > tempCounter - 1000) {
                // console.log('short: ' + (openTime - tempCounter));
                return;
            }
```

The combined effect is to see when select2 has opened, and if the user or the browser tries to close it less than 1 second after opening, prevent it.

A bit of a hack, but for us it makes the operation of select2 more predictable.

**Comment by edu2004eu** [comment]
One year and a few versions later, this still has no explanation to what can cause this.

A few more notes:
- it seems that if you tap on the select in the lower half, it doesn't repro (at least for me) -- this is consistent
- it seems that if I comment out https://github.com/select2/select2/blob/6633cda9061d9fc65075dd140c63f9a6adbcbd72/select2.js#L1422 the issue is fixed, so that might be the call to the `close` method that gets called involuntarily

**Comment by kevin-brown** [comment]
Fun fact: This appears to be fixed in 4.0.

I'd still love to know exactly why it's happening with the mask, but my money is still on a delayed `click` event that is being triggered on the mask.

**Comment by Dmitry9999** [comment]
@edu2004eu  Thanks for the solution, it helped to start with something. However, sometimes delay was more than 500ms (at least during debugging) and increasing value lead to situations when user could not quickly open and close a dropdown. So, after a little bit of further investigation and debugging I found out that somehow **"mousedown"** event is fired on the mask, right after user clicks on the <select> element (*not* on the mask). I did not find explicit "mousedown" call in the code, maybe @kevin-brown will have a clue. By the way, debugging on mobile phones is much easier with https://developer.chrome.com/devtools/docs/remote-debugging (in case you did not know).

To the point, in order not to affect touch events that are done by users, I changed the code
```if(!supportsTouchEvents || e.type != 'mousedown' || openTime < tempCounter - MIN_OPEN_TIME) {
    self.close();
}```

and maybe it can even be changed to
`if(!supportsTouchEvents || e.type != 'mousedown')`
if mousedown events cannot be fired on touch devices.
